### ISSUE TITLE
When the "Enable Publish Dates" is on AND the content's publish date is in the future and the content is in r

### ISSUE THREAD

**Issue by chrisny454** [transformstudios/statamic-review#52]
When I have this turned on....
![Screenshot 2024-11-25 120405](https://github.com/user-attachments/assets/0163564b-bbcd-4ea3-9543-180c2c00eec6)

And the content is scheduled to be in the future but is still in draft mode....
![Screenshot 2024-11-25 120549](https://github.com/user-attachments/assets/74cd6f28-7642-432c-b9b2-fc2045b210cb)

I get a 404 error and the review link does not work.

As soon as I change the date to be in the past the review link works again. Is there something I am doing wrong?
Fixing this functionality is crucial so that our staff can review content that is going to be published in the future otherwise the plugin is not going to meet our needs.

Let me know next steps, I am happy to test things out if you give me a place to start.

**Comment by edalzell** [comment]
@chrisny454 you're on the most recent version of this, v5.0?

**Comment by edalzell** [comment]
Hmm, I can't replicate this

**Comment by edalzell** [comment]
@chrisny454 anything unusual in your set up?

**Comment by chrisny454** [comment]
I believe I have a working solution for my needs and it may help someone else. I had to tweak your prepEntry() function first if statement like below:
<img width="656" alt="image" src="https://github.com/user-attachments/assets/74c4c77e-2f5c-4295-b245-3136d5d700fc" />
`
 private function prepEntry(Entry $entry): Entry
    {
        if ($entry->collection()->dated() && $entry->date()->isFuture()) {
            if($entry->collection()->futureDateBehavior() === 'private' )
            {
                //when futureDateBehavior is set to private, Statamic denies access so an override is needed.
                $entry->collection()->futureDateBehavior('public');
            }
            if(strpos($entry->collection()->routes(),'{month}') || strpos($entry->collection()->routes(),'{day}') || strpos($entry->collection()->routes(),'{year}'))
            {
                //do not change the date because the uri is dynamically created by the entry date.
            }
            else
            {
                $entry->date(now());
            }            
        }
        //if the revision changes the slug, the url is wrong and the review link 404s
        $originalSlug = $entry->slug();
        if ($entry->hasWorkingCopy()) {
            $entry = $entry->fromWorkingCopy();
        }        
        $entry->slug($originalSlug);
        return $entry->published(true);
    }
`

**Comment by edalzell** [comment]
> So one thing that may be working against us is that our entry URLS are generated based on the date they were / are going to be published. For example our Collections URL definition:
/blog/{year}/{month}/{day}/{slug}

This will be the issue for sure, cuz I set the date.

Nice fine, I'll get this fixed.

**Comment by edalzell** [comment]
The actual issue is that your post is published. If you have in draft the existing code works perfectly.

**Comment by chrisny454** [comment]
I guess its a technicality. Statamic calls posts that are published for future dates "scheduled" so they are technically not yet published and accessible. Our content loading process is to paste in the content and "schedule" the post for publishing in a future date. BUT before the date of the post going live, there are a couple people who need to review the content as a final check. Most of the time the content is loaded properly and nothing needs to be changed, but there are times where a misspelled word or a graphic is out of place causes the content to need updating after it has been "scheduled" and the review links have been used to proof the content one last time. I can see both use cases - yours and mine, but this process is one that has been in our workflows for years (we use to be on Sitefinity) and if I dont have to change it, I would rather not just to keep the peace.

**Comment by edalzell** [comment]
> I guess its a technicality. Statamic calls posts that are published for future dates "scheduled" so they are technically not yet published and accessible. Our content loading process is to paste in the content and "schedule" the post for publishing in a future date. BUT before the date of the post going live, there are a couple people who need to review the content as a final check. Most of the time the content is loaded properly and nothing needs to be changed, but there are times where a misspelled word or a graphic is out of place causes the content to need updating after it has been "scheduled" and the review links have been used to proof the content one last time. I can see both use cases - yours and mine, but this process is one that has been in our workflows for years (we use to be on Sitefinity) and if I dont have to change it, I would rather not just to keep the peace.

Makes sense, nice feature request (as this is not a bug, per se). Adding now.

**Comment by edalzell** [comment]
Interesting, I can't reproduce your issue. I have a dated collection, with the date in the route, and a published future post and I can preview it just fine.

@chrisny454 are you able to send me the site/entry so I can give it a try? If it's private you can send it to me via discord?

Also, can you show me the output of `php please support:details`?

**Comment by chrisny454** [comment]
Here is the output of php please support:details - the site is not yet live so I can't get you access to the login. I will try to fiddle with it and see if I can get some additional information on when the review links work or dont work. We arent live with the site for another 30 days.

Environment
Application Name: APP
Laravel Version: 11.36.1
PHP Version: 8.3.9
Composer Version: 2.8.2
Environment: local
Debug Mode: OFF
URL: <redacted>
Maintenance Mode: OFF
Timezone: America/New_York
Locale: en

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: NOT CACHED

Drivers
Broadcasting: log
Cache: file
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 4
Sites: 2 (English, Spanish)
Stache Watcher: Enabled
Static Caching: Disabled
Version: 5.45.0 PRO

Statamic Addons
duncanmcclean/cookie-notice: 9.1.0
jacksleight/statamic-bard-texstyle: 3.4.1
statamic/eloquent-driver: 4.19.1
transformstudios/review: 5.0

Statamic Eloquent Driver
Asset Containers: file
Assets: eloquent
Blueprints: file

**Comment by edalzell** [comment]
> the site is not yet live so I can't get you access to the login.

I don’t want access to your live site, I’d like a copy of the site so I can see the issue.

Only difference is eloquent. I’ll try that and see if it makes a difference.
